grails-app/services/org/pih/warehouse/data/ProductSupplierAttributeDataService.groovy

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) 2012 Partners In Health.  All rights reserved.
+ * The use and distribution terms for this software are covered by the
+ * Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+ * which can be found in the file epl-v10.html at the root of this distribution.
+ * By using this software in any fashion, you are agreeing to be bound by
+ * the terms of this license.
+ * You must not remove this notice, or any other, from this software.
+ **/
+package org.pih.warehouse.data
+
+import org.pih.warehouse.core.EntityTypeCode
+import org.pih.warehouse.core.UnitOfMeasure
+import org.pih.warehouse.importer.ImportDataCommand
+import org.pih.warehouse.product.Attribute
+import org.pih.warehouse.product.Product
+import org.pih.warehouse.product.ProductAttribute
+import org.pih.warehouse.product.ProductSupplier
+
+class ProductSupplierAttributeDataService {
+
+    Boolean validate(ImportDataCommand command) {
+        log.info "Validate data " + command.filename
+        command.data.eachWithIndex { params, index ->
+
+            def productCode = params.productCode
+            def productSupplierCode = params.productSupplierCode
+            def attributeCode = params.attributeCode
+            def attributeValue = params.attributeValue
+            def unitOfMeasure = params.unitOfMeasure
+
+            // Product code is required
+            if (!productCode) {
+                command.errors.reject("Row ${index + 1}: Product code is required")
+                return
+            }
+
+            // Product should exist
+            Product product = Product.findByProductCode(productCode)
+            if (!product) {
+                command.errors.reject("Row ${index + 1}: Product with code: ${productCode} does not exist")
+                return
+            }
+
+            // Product supplier code is required
+            if (!productSupplierCode) {
+                command.errors.reject("Row ${index + 1}: Product source code is required")
+                return
+            }
+
+            // Product supplier should exist
+            ProductSupplier productSupplier = ProductSupplier.findByCode(productSupplierCode)
```

Comment:
Let's ask Kelsey about this. I know I probably said there should be validation but now I'm wondering if it makes sense to let a user assign their own when creating. We obviously can't use this field to modify the productSupplier.code unless we add productSupplier.id to the import.

Reply:
@jmiranda Sure, I will ask it on today's call. However, in my understanding, this should be required and should exist, because this is the import of product attributes related to the product supplier (otherwise we should include the creation of product supplier if the product supplier code is not provided, but ticket's acceptance criteria did not specified creation of product supplier)